modules/core/src/test/java/org/apache/ignite/internal/util/IgniteUtilsSelfTest.java

```diff
@@ -1455,6 +1470,51 @@ public void testUncompressedSize() throws Exception {
         }
     }
 
+    /**
+     * Test to verify the {@link U#acquireAndExecute(IgniteSemaphore, IgniteCallable, int)}.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testAcquireAndExecute() throws Exception {
```

Comment:
You can also add a negative scenario, e.g. what will happen if there is an error during callable's execution